swarm/pss/snapshot_test.go

```diff
@@ -0,0 +1,433 @@
+	nodeAddrs = make(map[enode.ID][]byte)
+	recipients = make(map[int][]enode.ID)
+	allowed = make(map[int][]enode.ID)
+	expectedMsgs = make(map[enode.ID][]uint64)
+	allowedMsgs = make(map[enode.ID][]uint64)
+	senders = make(map[int]enode.ID)
+	handlerC = make(chan handlerNotification)
+	doneC = make(chan struct{})
+	errC = make(chan error)
+	msgC = make(chan handlerNotification)
+}
+
+func isDone() bool {
+	mu.Lock()
+	defer mu.Unlock()
+	return handlerDone
+}
+
+func setDone() {
+	mu.Lock()
+	defer mu.Unlock()
+	handlerDone = true
+}
+
+func getCmdParams(t *testing.T) (int, int) {
+	args := strings.Split(t.Name(), "/")
+	msgCount, err := strconv.ParseInt(args[1], 10, 16)
+	if err != nil {
+		t.Fatal(err)
+	}
+	nodeCount, err := strconv.ParseInt(args[2], 10, 16)
+	if err != nil {
+		t.Fatal(err)
+	}
+	return int(msgCount), int(nodeCount)
+}
+
+func readSnapshot(t *testing.T, nodeCount int) simulations.Snapshot {
+	f, err := os.Open(fmt.Sprintf("testdata/snapshot_%d.json", nodeCount))
+	if err != nil {
+		t.Fatal(err)
+	}
+	jsonbyte, err := ioutil.ReadAll(f)
+	if err != nil {
+		t.Fatal(err)
+	}
+	var snap simulations.Snapshot
+	err = json.Unmarshal(jsonbyte, &snap)
+	if err != nil {
+		t.Fatal(err)
+	}
+	return snap
+}
+
+func assingTestVariables(sim *simulation.Simulation, msgCount int) {
+	log.Debug("-------------------------------------------------------------------------")
+	for _, nodeId := range sim.NodeIDs() {
+		nodeAddrs[nodeId] = nodeIDToAddr(nodeId)
+	}
+
+	for i := 0; i < int(msgCount); i++ {
+		msgAddr := pot.RandomAddress() // we choose message addresses randomly
+		msgs = append(msgs, msgAddr.Bytes())
+		smallestPo := 256
+		var targets []enode.ID
+		var prev int
+
+		// loop through all nodes and add the message to recipient indices
+		for _, nod := range sim.Net.GetNodes() {
+			po, _ := pof(msgs[i], nodeAddrs[nod.ID()], 0)
+			depth := kademlias[nod.ID()].NeighbourhoodDepth()
+
+			// only nodes with closest IDs (wrt msg) will receive the msg
+			if po > prev {
+				prev = po
+				targets = nil
+				targets = append(targets, nod.ID())
+			} else if po == prev {
+				targets = append(targets, nod.ID())
+			}
+
+			if po >= depth {
+				maxMessages++
+				allowed[i] = append(allowed[i], nod.ID())
+				allowedMsgs[nod.ID()] = append(allowedMsgs[nod.ID()], uint64(i))
+			}
+
+			// a node with the smallest PO (wrt msg) will be the sender
+			if po < smallestPo {
+				smallestPo = po
+				senders[i] = nod.ID()
+			}
+		}
+
+		msgsToReceive += len(targets)
+		for _, id := range targets {
+			recipients[i] = append(recipients[i], id)
+			expectedMsgs[id] = append(expectedMsgs[id], uint64(i))
+		}
+
+		log.Debug("nn for msg", "targets", len(recipients[i]), "msgidx", i, "msg", common.Bytes2Hex(msgAddr[:8]), "sender", senders[i], "senderpo", smallestPo)
+	}
+	log.Debug("msgs to receive", "count", msgsToReceive)
+}
+
+func TestProxNetwork(t *testing.T) {
+	if (*runNodes > 0 && *runMessages == 0) || (*runMessages > 0 && *runNodes == 0) {
```

Comment:
set proper defaults, no need to check

Reply:
this is needed in case of invalid explicit input

Reply:
I think it's  better to add the flag in the actual test itself, since these flags are only relevant for this specific test - unlike the `pss_test.go` flags, which apply to the full package test. Keep in mind you will need `flag.Parse()` too in that case (you get it for free now with `pss_test.go:init()`